### PR TITLE
Featured Badge fixes

### PR DESCRIPTION
**Pull request by sparrowDom**
### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- ~[ ] Wrap any new text/strings for translation~
- ~[ ] Map any new environment variables with a default value in the Webpack config~
- ~[ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/docs)~

### Description:
- show featured listings only if a listing is featured an no other badges are shown
- show featured badge on search results

### CODE REVIEW

**Comment by sparrowDom** [comment]
The fix for showing featured badge on search results is not obvious from the code diff because of the renames. The important non changed line (but logically changed because of the renames) is src/components/listings-grid.js:102

File: src/components/listing-card.js [hunk 1]
```diff
@@ -89,7 +89,9 @@ class ListingCard extends Component {
       o => offerStatusToListingAvailability(o.status) === 'sold'
     )
     const isWithdrawn = status === 'inactive'
-    const showFeaturedBadge = this.props.featured && !isSold && !isPending
+    const showPendingBadge = isPending && !isWithdrawn
+    const showSoldBadge = isSold || isWithdrawn
```
Comment (franckc) [inline src/components/listing-card.js]: Do we actually want to show the Sold badge on withdrawn items ?
Would it rather make sense to hide those listings ? Though if it complicates the logic not worth it - we can implement that in the back-end when we transition to fetching listing IDs from back-end next sprint.

Comment (franckc) [reply src/components/listing-card.js]: Yes, makes sense. Not worth adding complexity.

**Comment by franckc** [comment]
@micahalcorn ok with this fix ?

**Comment by micahalcorn** [comment]
@franckc yes, I've added some changes to handle the badges on the listing detail as well. ✔️

**Pull request merged**